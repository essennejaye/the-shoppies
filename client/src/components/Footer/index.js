import React from 'react';

const Footer = () => {
  let currentYear = new Date().getFullYear();
  return (
    <section className='footer'>
      <div className="name-date">
        <p>
          Essennejaye &copy; <span data-testid='currentYear'>{currentYear}</span>
        </p>
      </div>
      <div className="icon-img">
        <a href="https://github.com/essennejaye"><img src="../images1/github-32px.png" alt="github-icon" /></a>
        <a href="https://www.linkedin.com/in/satalia-n-jefferson"><img src="../images1/linkedin-32px.png" alt="linkedin-icon" /></a>
      </div>
    </section>
      )
  }
export default Footer;
