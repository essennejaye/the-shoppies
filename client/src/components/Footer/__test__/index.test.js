import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Footer from "..";

const getCurrentYear = () => new Date();

describe("Footer component", () => {
  it("renders", () => {
    render(<Footer currentYear={getCurrentYear} />);
  });

  it("matches snapshot DOM node structure", () => {
    const { asFragment } = render(<Footer currentYear={getCurrentYear} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("inserts name, copyright symbol", () => {
    render(<Footer currentYear={getCurrentYear} />);
    expect(screen.getByText(/Essennejaye Â©/)).toBeInTheDocument();
  });

  it("inserts current year", () => {
    render(<Footer />);
    expect(screen.getByTestId("currentYear")).toHaveTextContent(
      `${getCurrentYear().getFullYear()}`
    );
  });

  it("inserts icons", () => {
    render(<Footer currentYear={getCurrentYear} />);
    const githubIconImg = screen.getByAltText("github-icon");
    const linkedinIconImg = screen.getByAltText("linkedin-icon");
    expect(githubIconImg).toBeInTheDocument();
    expect(linkedinIconImg).toBeInTheDocument();
  });
  
  it("renders links", () => {
    render(<Footer currentYear={getCurrentYear} />);
    const githubLink = screen.getByRole("link", { name: "github-icon" });
    const linkedinLink = screen.getByRole("link", { name: "linkedin-icon" });
    expect(githubLink).toHaveAttribute ('href', "https://github.com/essennejaye");
    expect(linkedinLink).toHaveAttribute('href', 'https://www.linkedin.com/in/satalia-n-jefferson');
  });
});
