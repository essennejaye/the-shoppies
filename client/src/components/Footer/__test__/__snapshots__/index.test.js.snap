// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="footer"
  >
    <div
      class="name-date"
    >
      <p>
        Essennejaye Â© 
        <span
          data-testid="currentYear"
        >
          2021
        </span>
      </p>
    </div>
    <div
      class="icon-img"
    >
      <a
        href="https://github.com/essennejaye"
      >
        <img
          alt="github-icon"
          src="../images1/github-32px.png"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/satalia-n-jefferson"
      >
        <img
          alt="linkedin-icon"
          src="../images1/linkedin-32px.png"
        />
      </a>
    </div>
  </section>
</DocumentFragment>
`;
